src/serial.rs

```diff
@@ -567,13 +575,25 @@ mod tests {
     }
 
     #[derive(Default)]
-    struct ExampleSerialMetrics {
+    struct ExampleSerialEvents {
         read_count: AtomicU64,
         out_byte_count: AtomicU64,
         tx_lost_byte_count: AtomicU64,
+        buffer_ready_event: Option<EventFd>,
+    }
+
+    impl ExampleSerialEvents {
+        fn new() -> Self {
+            ExampleSerialEvents {
+                read_count: AtomicU64::new(0),
+                out_byte_count: AtomicU64::new(0),
+                tx_lost_byte_count: AtomicU64::new(0),
+                buffer_ready_event: Some(EventFd::new(libc::EFD_NONBLOCK).unwrap()),
```

Comment:
I don't see the need to have `buffer_ready_event` as an option. Why can't we just always initialize it with `EventFd::new(libc::EFD_NONBLOCK).unwrap()`? This would simplify the code where `buffer_ready_event` is used, and where that field is not used, then having it initialize doesn't cause any problems.